### PR TITLE
[Indexer] Decoding of event values

### PR DESCRIPTION
**Pull request by sideninja**
Closes: #16 

## Description
This PR adds the decoding of event data into evm typed data that will be used for storage and querying. 

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: models/block.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		return nil, err
+	}
+
+	hashes := make([]common.Hash, len(b.TransactionHashes))
+	for i, h := range b.TransactionHashes {
+		hashes[i] = common.HexToHash(h.ToGoValue().(string))
+	}
+
+	return &types.Block{
+		ParentBlockHash:   common.HexToHash(b.ParentBlockHash),
```
Comment (m-Peter) [inline models/block.go:45]: The `Hash` field is not being passed here.

File: models/block_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+func Test_DecodeBlockExecutedEvent(t *testing.T) {
+	block := &types.Block{
+		ParentBlockHash: common.HexToHash("0x1"),
```
Comment (m-Peter) [inline models/block_test.go:14]: Here as well.

File: models/block.go (hunk 1 above)
Comment (sideninja) [reply models/block.go:45]: what do you mean?

Comment (m-Peter) [reply models/block.go:45]: Never mind, I just forgot that it is computed through a method: https://github.com/onflow/flow-go/blob/master/fvm/evm/types/block.go#L35-L39. I thought it was a field instead.

**Review by m-Peter — APPROVED** [review]
LGTM!

**Comment by sideninja** [comment]
@franklywatson this shouldn't be merged yet.

**Pull request merged**